segmentation_models_pytorch/fpn/model.py

```diff
@@ -36,7 +37,11 @@ def __init__(
             dropout=0.2,
             activation='sigmoid',
             final_upsampling=4,
+            decoder_merge_policy='add'
     ):
+        if decoder_merge_policy not in ['add', 'cat']:
+            raise ValueError("`merge_policy` must be one of: ['add', 'cat']")
+            
```

Comment:
I think this is responibility of decoder to know what kind of merge policy to use, however in this case we first build encoder and load its weights which is time and memory consuming operation.. what do you think? is it it better to move this to decoder?

Reply:
> however in this case we first build encoder and load its weights which is time and memory consuming 

Hm, you mean that it can be annoying for library user to know that he passed wrong merge_policy after waiting for 20 minutes for weights downloading? :) What do you think if move this check to the utility function and will do such checks in both decoder and model?

Reply:
Just added `check_merge_policy` function. Please see and let me know what do you think about such approach.